modules/firmware_apps/app_store.py

```diff
@@ -255,6 +255,20 @@ def update(self, delta):
             self.prepare_available_menu()
         elif self.state == "installed_menu" and not self.installed_menu:
             self.prepare_installed_menu()
+        elif self.state == "refreshing_index":
+            try:
+                self.response = await async_helpers.unblock(
+                    get, render_update, APP_STORE_LISTING_URL
+                )
+            except Exception:
+                self.update_state("no_index")
+            self.update_state("index_received")
```

Comment:
Yes, per Naomi's comment I think this ought to be like this?

```suggestion
            try:
                self.response = await async_helpers.unblock(
                    get, render_update, APP_STORE_LISTING_URL
                )
                self.update_state("index_received")
            except Exception:
                self.update_state("no_index")
```